File: algorithms/Analysis/Anagram/anagramSolution3.py
def anagramSolution3(s1, s2):
    c1 = [0]*26
    c2 = [0]*26

    for i in range(len(s1)):
        pos = ord(s1[i]) - ord('a')
        c1[pos] = c1[pos] + 1

    for i in range(len(s2)):
        pos = ord(s2[i]) - ord('a')
        c2[pos] = c2[pos] + 1

    j = 0;
    ok = True
    while j < 26 and ok:
        if (c1[j] == c2[j]):
            j += 1
        else:
            ok = False

    return ok

print(anagramSolution3('abcd', 'bcda'))

File: algorithms/Analysis/Anagram/anagramSolution3.go
package main

import (
	"fmt"
)

func main() {
	s1 := "abcd"
	s2 := "bcda"

	fmt.Println(AnagramSolution3(s1, s2))
}

func AnagramSolution3(s1 string, s2 string) bool {
	var c1 [26]int
	var c2 [26]int

	for _, w1 := range s1 {
		pos := int(w1) - int('a')
		c1[pos] = 1
	}

	for _, w2 := range s2 {
		pos := int(w2) - int('a')
		c2[pos] = 1
	}

	j := 0
	ok := true
	for j < 26 && ok {
		if c1[j] == c2[j] {
			j++
		} else {
			ok = false
		}
	}

	return ok
}
